src/sandbox/sandbox-state-methods.js

```diff
@@ -123,6 +123,8 @@ Oskari.clazz.category('Oskari.Sandbox', 'state-methods', {
             // reset to the default state
             bundle.setState(bundleState);
         }
+        // trigger an event letting bundles know that the whole UI has changed to clean functionalities
+        this.notifyAll(Oskari.eventBuilder('UIChangeEvent')('sandbox'));
```

Comment:
This should be sent before calling setState() on bundles since it's mainly for cleanup. This way if setState() causes a popup to open it's not cleared right away because of this.